src/bpftrace.cpp

```diff
@@ -1493,8 +1494,8 @@ std::optional<std::string> BPFtrace::get_watchpoint_binary_path() const
     auto args = split_string(cmd_, ' ', /* remove_empty */ true);
     assert(!args.empty());
     return resolve_binary_path(args[0]).front();
-  } else if (pid())
-    return "/proc/" + std::to_string(pid()) + "/exe";
+  } else if (pid().has_value())
+    return "/proc/" + std::to_string(pid().value_or(0)) + "/exe";
```

Comment:
No need for the `.value_or(0)` part IMO, since `pid()` must already have value.
```suggestion
    return "/proc/" + std::to_string(*pid()) + "/exe";
```

Reply:
Makes sense.